packages/teuchos/parameterlist/src/Teuchos_ParameterListModifier.hpp

```diff
@@ -0,0 +1,243 @@
+   *
+   *  \param parameter_template_name [in] The name of the parameter template in \a valid_pl
+   *
+   *  \param pl [in] The parameter list that needs to be validated
+   *
+   *  \param valid_pl [in,out] The parameter list that is being used as a template for validation
+   *
+   *  \param exclude_parameters [in] An optional list of parameter names to exclude
+   *
+   */
+  int expandParameters(const std::string &parameter_template_name, ParameterList &pl,
+      ParameterList &valid_pl,
+      const Array<std::string> &exclude_parameters=Array<std::string>()) const;
+
+
+  /** \brief Create sublists in the valid parameter list using a template parameter from the valid
+   *  parameter list and the names of sublists in the list being validated.
+   *
+   *  \param sublist_template_name [in] The name of the sublist template in \a valid_pl
+   *
+   *  \param pl [in] The parameter list that needs to be validated
+   *
+   *  \param valid_pl [in,out] The parameter list that is being used as a template for validation
+   *
+   *  \param exclude_sublists [in] An optional list of sublist names to exclude
+   *
+   */
+  int expandSublists(const std::string &sublist_template_name, ParameterList &pl,
+      ParameterList &valid_pl,
+      const Array<std::string> &exclude_sublists=Array<std::string>()) const;
+
+
+  int setDefaultsInSublists(const std::string &param_name, ParameterList &pl,
+      const Array<std::string> &sublist_names) const;
+
+
+  /** \brief Create sublists in the valid parameter list using a base name and the corresponding sublists
+   *  in the parameter list being validated.
+   *
+   *  \param base_name [in] The root name of the sublists to look for and create
+   *
+   *  \param pl [in] The parameter list that needs to be validated
+   *
+   *  \param valid_pl [in,out] The parameter list that is being used as a template for validation
+   *
+   *  \param allow_base_name [in] Allow the parameter list \a pl to contain a parameter with the same name
+   *    as base_name.
+   *
+   */
+  int expandSublistsUsingBaseName(const std::string &base_name, ParameterList &pl,
+      ParameterList &valid_pl, const bool &allow_base_name = true) const;
+
+
+  /** \brief Create an array parameter from a scalar parameter
+   *
+   *  \param param_name [in] The parameter name of a scalar
+   *
+   *  \param new_name [in] The new parameter name of the array containing the scalar
+   *
+   *  \param pl [in,out] Allow the parameter list \a pl to contain a parameter with the same name
+   *    as base_name.
+   *
+   *  \param throw_if_new_name_exists [in] Throw an error if the new name already exists in \a pl
+   *
+   *  This currently works with string, integer, and float scalars.
+   */
+  template<typename T>
+  bool replaceScalarParameterWithArray(const std::string &param_name, const std::string &new_name,
+      ParameterList &pl, const bool &throw_if_new_name_exists = true) const;
+
+
+protected:
+  std::string name_;
```

Comment:
Write `std::string name_ ("ANONYMOUS");`, so you can use `=default` on the default constructor.